src/fsFile.ts

```diff
@@ -0,0 +1,145 @@
+import * as fs from 'fs';
+import { extToMime } from 'ardrive-core-js';
+import { basename, join } from 'path';
+import { Bytes, DataContentType } from './types';
+import { BulkFileBaseCosts, FolderUploadBaseCosts } from './ardrive';
+
+type BaseFileName = string;
+type FilePath = string;
+
+/**
+ *  Fs + Node implementation file size limitations -- tested on MacOS Sep 27, 2021
+ *
+ *  Public : 2147483647 bytes
+ *  Private: 2147483646 bytes
+ */
+const maxFileSize: Bytes = 2147483646;
+
+export interface FileInfo {
+	dataContentType: DataContentType;
+	lastModifiedDateMS: number;
+	fileSize: Bytes;
+}
+
+/**
+ * Reads stats of a file or folder  and constructs a File or Folder wrapper class
+ *
+ * @remarks import and use `isFolder` type-guard to later determine whether a folder or file
+ *
+ * @example
+ *
+ * const fileOrFolder = wrapFileOrFolder(myFilePath);
+ *
+ * if (isFolder(fileOrFolder)) {
+ * 	// Type is: Folder
+ * } else {
+ * 	// Type is: File
+ * }
+ *
+ */
+export function wrapFileOrFolder(fileOrFolderPath: FilePath): FsFile | FsFolder {
+	const entityStats = fs.statSync(fileOrFolderPath);
+
+	if (entityStats.isDirectory()) {
+		return new FsFolder(fileOrFolderPath, entityStats);
+	}
+
+	return new FsFile(fileOrFolderPath, entityStats);
+}
+
+/** Type-guard function to determine if returned class is a File or Folder */
+export function isFolder(fileOrFolder: FsFile | FsFolder): fileOrFolder is FsFolder {
+	return Object.keys(fileOrFolder).includes('files') || Object.keys(fileOrFolder).includes('folders');
+}
+
+export class FsFile {
+	constructor(public readonly filePath: FilePath, public readonly fileStats: fs.Stats) {
+		if (this.fileStats.size >= maxFileSize) {
+			throw new Error(`Files greater than "${maxFileSize}" bytes are not yet supported!`);
+		}
+	}
+
+	baseCosts?: BulkFileBaseCosts;
```

Comment:
So we need these as optional because we are feeding the value with recursion, there's no easy & clean way to put it in the constructor, right?
That's fine as long as it's hidden behind the getter

Reply:
To put these base cost logic inside of the constructor, we'd mainly need to be able access to the price estimator in the layer that we would wrap the file. 

In the code's current state, I've lifted the file wrapping out into the commander layer so we can determine whether we're performing a bulk upload or a single file upload before proceeding into the ArDrive layer.

Making it optional and hiding it behind the getter was my quick solution as I fed the values through the recursive estimation method.